R/word_splitters.R

```diff
@@ -10,47 +43,57 @@
 #' to the actual form of the ending) and morphemes (typically denoted in
 #' Wiktionary by etymology templates).
 #'
-#' @param word Character; a word to process.
+#' @inheritParams process_word
+#' @param current_depth Integer; current recursion depth.
+#' @param max_depth Integer; maximum recursion depth.
 #'
 #' @return Character; the word split into pieces.
 #' @export
 #' @examples
-process_word <- function(word) {
-  changed <- TRUE
-  endings <- character(0)
-  # first, do endings repeatedly till no change
-  current_word <- word
-  while(changed) {
-    inf_break <- split_inflections(current_word)
-    if (identical(inf_break, current_word) |
-        length(inf_break) == 0) { # zero length if word (piece) not found
-      changed <- FALSE
-    } else {
-      # names are "safe" to use as keys at this point.
-      current_word <- inf_break[["base_word"]]
-      # current_word <- inf_break[[1]]
-      endings[length(endings) + 1] <- inf_break[["ending"]]
-      # endings[length(endings) + 1] <- inf_break[[2]]
-    }
+process_word_recursive <- function(word, current_depth = 1, max_depth = 30) {
+  if (current_depth > max_depth) {
+    message("maximum recursion depth of ", max_depth, " reached.")
+    return(word)
+  }
+  inflection_changed <- TRUE
+  morpheme_changed <- TRUE
+  ending <- character(0)
+  # first, do endings repeatedly till no change. Nope, just do once, then
+  # recurse. I think.
```

Comment:
I'm just removing the comment without making an issue, because I think this is the right way to go. Unless you have an argument for processing multiple inflection endings per call here?

The only notable difference that I'm aware of will be in the resulting names. There are examples of words (e.g. "ending") that are marked both as having an "ing" inflection *and* as having an "-ing" suffix, in separate templates. So whether the "ing" gets tagged as a suffix or an inflection depends on what we look for first. Currently, we look for a single inflection, then a single morpheme split, per recursion. So a word like "endings" gets the "s" stripped off as an inflection, then the "ing" stripped off as a prefix. But in the old way, where we process multiple inflections before looking for morpheme splits, both the "ing" and the "s" get processed as inflections. 

We could also recover the old behavior by starting a new recursive call after stripping off the first inflection, rather than doing (maybe) two things in one call.

Hmm, I've convinced myself that there should be an issue here after all. :)